libs/ui-lib/lib/ocm/components/clusterConfiguration/ExternalPlatformDropdown.tsx

```diff
@@ -60,17 +67,78 @@ export const externalPlatformTypes: Record<ExternalPlatformType, ExternalPlatfor
 
 export const ExternalPlatformDropdown = ({
   showOciOption,
-  disabledOciTooltipContent,
-  isOciDisabled,
   onChange,
-  dropdownIsDisabled,
+  featureSupportLevelData,
+  cpuArchitecture,
+  clusterExists,
+  clusterPlatform,
 }: ExternalPlatformDropdownProps) => {
   const [field, { value }, { setValue }] = useField<string>(INPUT_NAME);
   const [isOpen, setOpen] = React.useState(false);
+
+  const newFeatureSupportLevelContext = useNewFeatureSupportLevel();
+
+  const isNutanixDisabled = newFeatureSupportLevelContext.isFeatureDisabled(
+    'NUTANIX_INTEGRATION',
+    featureSupportLevelData,
+  );
+
+  const nutanixDisabledReason = newFeatureSupportLevelContext.getFeatureDisabledReason(
+    'NUTANIX_INTEGRATION',
+    featureSupportLevelData,
+    cpuArchitecture,
+  );
+
+  const isVsphereDisabled = newFeatureSupportLevelContext.isFeatureDisabled(
+    'VSPHERE_INTEGRATION',
+    featureSupportLevelData,
+  );
+
+  const vsphereDisabledReason = newFeatureSupportLevelContext.getFeatureDisabledReason(
+    'VSPHERE_INTEGRATION',
+    featureSupportLevelData,
+    cpuArchitecture,
+  );
+
+  const isOracleDisabled = newFeatureSupportLevelContext.isFeatureDisabled(
+    'EXTERNAL_PLATFORM_OCI',
+    featureSupportLevelData,
+  );
+
+  const oracleDisabledReason = newFeatureSupportLevelContext.getFeatureDisabledReason(
+    'EXTERNAL_PLATFORM_OCI',
+    featureSupportLevelData,
+    cpuArchitecture,
+  );
```

Comment:
This is not going to scale well. Consider mapping the different platform integrations to something like

```
type IntegrationStateMap = {
  [key in FeatureId]: {
    isDisabled: boolean;
    isDisabledReason?: string;
  };
};
```
and using that in the ExternalPlatformDropdown.

Reply:
done!

Reply:
Sorry, your followup to my comment resulted in even worse-scaling code. I was hoping you would do something like [this](https://github.com/openshift-assisted/assisted-installer-ui/blob/master/libs/ui-lib/lib/ocm/components/clusterConfiguration/review/ReviewOperatorsTable.tsx#L18-L31) and use the resulting map in the rest of the ExternalPlatformDropdown.

You could then do things like 
```
stateMap.map(({platform: state}) => state.isDisabled).every(Boolean)
```
to see whether there are any non-disabled platforms instead of

```
  const dropdownIsDisabled =
    (clusterExists && clusterPlatform === 'oci') ||
    (nutanixInfoDisabled.isDisabled &&
      vsphereInfoDisabled.isDisabled &&
      oracleInfoDisabled.isDisabled);
```

And there are probably more places in your code where you manually specify each and every platform to check something which just seems unnecessary, prone to mistakes, and badly scaling.

Reply:
ok, I'll try to improve my code.